components/Collections/src/MDCCollectionInfoBarView.m

```diff
@@ -0,0 +1,261 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCCollectionInfoBarView.h"
+
+#import "MaterialShadowLayer.h"
+#import "MaterialTypography.h"
+
+const CGFloat MDCCollectionInfoBarAnimationDuration = 0.3f;
+
+static const CGFloat kInfoBarDefaultHeight = 48.0f;
+static const NSTimeInterval kInfoBarDefaultDismissal = 1.0f;
+static const CGFloat kInfoBarDefaultLabelHorizontalPadding = 16.0f;
+
+// Colors derived from http://www.google.com/design/spec/style/color.html#color-color-palette .
+static const uint32_t kInfoBarBlueColor = 0x448AFF;  // Blue A200
+static const uint32_t kInfoBarRedColor = 0xF44336;   // Red 500
+
+// Creates a UIColor from a 24-bit RGB color encoded as an integer.
+static inline UIColor *ColorFromRGB(uint32_t rgbValue) {
+  return [UIColor colorWithRed:((CGFloat)((rgbValue & 0xFF0000) >> 16)) / 255
+                         green:((CGFloat)((rgbValue & 0x00FF00) >> 8)) / 255
+                          blue:((CGFloat)((rgbValue & 0x0000FF) >> 0)) / 255
+                         alpha:1];
+}
+
+@interface ShadowedView : UIView
+@end
+
+@implementation ShadowedView
++ (Class)layerClass {
+  return [MDCShadowLayer class];
+}
+@end
+
+@implementation MDCCollectionInfoBarView {
+  CGFloat _backgroundTransformY;
+  CALayer *_backgroundBorderLayer;
+  UITapGestureRecognizer *_tapGesture;
+}
+
+- (instancetype)initWithStyle:(MDCInfoBarStyle)style
+                         kind:(MDCInfoBarKind)kind
+               collectionView:(UICollectionView *)collectionView {
+  _style = style;
+  _kind = kind;
+
+  // Determine frame based on if this is a header or footer.
+  CGFloat offsetY = 0;
+  if (_kind == MDCInfoBarKindFooter) {
+    offsetY = collectionView.bounds.size.height - kInfoBarDefaultHeight;
+  }
+  CGRect frame = CGRectMake(0, offsetY, collectionView.bounds.size.width, kInfoBarDefaultHeight);
+
+  self = [super initWithFrame:frame];
```

Comment:
This is typically the first line of the initializer.
Everything else goes inside the if clause.